mediapipe/calculators/ovms/openvinoinferencedumputils.cc

```diff
@@ -0,0 +1,229 @@
+//*****************************************************************************
+// Copyright 2023 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <ctime>
+#include <chrono>
+#include <filesystem>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <openvino/openvino.hpp>
+
+#include "mediapipe/calculators/ovms/openvinoinferencedumputils.h"
+
+namespace mediapipe {
+    
+std::unordered_map<std::string, int> dumpCounters;
+
+using InferenceInput = std::map<std::string, ov::Tensor>;
+
+static std::stringstream dumpOvTensor(const ov::Tensor& tensor) {
+    std::stringstream dumpStream;
+    switch (tensor.get_element_type()) {
+        // The cosine window and square root of Hann are equivalent.
+        case ov::element::Type_t::f64: {
+            const _Float64* input_tensor_access = reinterpret_cast<_Float64*>(tensor.data());
+            dumpStream << " tensor: [ ";
+            for (int x = 0; x < tensor.get_size(); x++) {
+                dumpStream << &input_tensor_access[x] << " ";
+            }
+            dumpStream << " ]";
+            break;
+        }
```

Comment:
Why the `&`? Shouldn't it be `&input_tensor_access[x] --> input_tensor_access[x]`?
I would also just rename it to `value`.
```suggestion
#define TYPE_CASE(ENUM, TYPE) \
        case ENUM: { \
            const TYPE* input_tensor_access = reinterpret_cast<TYPE*>(tensor.data()); \
            dumpStream << " tensor: [ "; \
            for (int x = 0; x < tensor.get_size(); x++) { \
                dumpStream << &input_tensor_access[x] << " "; \
            } \
            dumpStream << " ]"; \
            break; \
        }
        TYPE_CASE(ov::element::Type_t::f64, _Float64*)
        TYPE_CASE(ov::element::Type_t::f32, _Float32*)
        ...
```
Then reuse for all.